screeps-game-api/src/macros.rs

```diff
@@ -93,40 +120,48 @@ macro_rules! get_from_js {
 }
 
 /// Macro used to encapsulate all screeps game objects
-/// 
-/// Macro syntax: 
-/// reference_wrapper!{
-///     $obj1,
-///     $obj2,
-///     ...
+///
+/// Macro syntax:
+///
+/// ```
+/// reference_wrapper! {
+///     #[reference(instance_of = "Creep")]
+///     Creep,
+///     #[reference(instance_of = "Room")],
+///     Room,
+///     // ...
 /// }
-/// 
+/// ```
+///
 /// Screeps game objects, in javascript, can be accessed via stdweb's `Reference`
 /// object. This macro: 
 ///   - Creates a struct named `objX`;
 ///   - Implements traits `AsRef<Reference>`, `TryFrom<Value>` for `objX`
```

Comment:
Macro doc will require some adjustments.

Reply:
Especially since the list of traits you get by deriving `ReferenceType` is far from obvious.
You get `AsRef<Reference>, `InstanceOf`, `TryFrom<Value>`, `TryFrom<Reference>, `[Try]From<T> for Reference/Value`...

Maybe just reference the doc for one of the object...

Reply:
Right, will fix this. We might as well list everything `ReferenceType` does as well as mention it's all done through `#[derive(ReferenceType)]`.